src/compiler/crystal/compiler.cr

```diff
@@ -18,6 +18,7 @@ module Crystal
   # optionally generates an executable.
   class Compiler
     CC = ENV["CC"]? || "cc"
+    CL = "cl"
```

Comment:
>The LINK tool prepends the options and arguments defined in the LINK environment variable and appends the options and arguments defined in the _LINK_ environment variable to the command line arguments before processing.

From this wording, it appears to me that the `LINK` env var is used to prepend options to the link command, not contain the path to the executable.